src/Amazon.ElasticBeanstalk.Tools/Commands/CreatePackageCommand.cs

```diff
@@ -0,0 +1,203 @@
+                this.GetStringValueOrDefault(this.ProjectLocation, CommonDefinedCommandOptions.ARGUMENT_PROJECT_LOCATION, false));
+
+            string configuration = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Configuration, CommonDefinedCommandOptions.ARGUMENT_CONFIGURATION, false) ?? "Release";
+            string targetFramework = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.TargetFramework, CommonDefinedCommandOptions.ARGUMENT_FRAMEWORK, false);
+            string publishOptions = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.PublishOptions, CommonDefinedCommandOptions.ARGUMENT_PUBLISH_OPTIONS, false);
+
+            if (string.IsNullOrEmpty(targetFramework))
+            {
+                targetFramework = Utilities.LookupTargetFrameworkFromProjectFile(projectLocation);
+                if (string.IsNullOrEmpty(targetFramework))
+                {
+                    targetFramework = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.TargetFramework, CommonDefinedCommandOptions.ARGUMENT_FRAMEWORK, true);
+                }
+            }
+
+                string application = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Application, EBDefinedCommandOptions.ARGUMENT_EB_APPLICATION, true);
+                string environment = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Environment, EBDefinedCommandOptions.ARGUMENT_EB_ENVIRONMENT, true);
+                string versionLabel = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.VersionLabel, EBDefinedCommandOptions.ARGUMENT_EB_VERSION_LABEL, false) ?? DateTime.Now.Ticks.ToString();
+
+            var dotnetCli = new DotNetCLIWrapper(this.Logger, projectLocation);
+
+            var publishLocation = Utilities.DeterminePublishLocation(null,  projectLocation, configuration, targetFramework);
+            this.Logger?.WriteLine("Determine publish location: " + publishLocation);
+
+
+            this.Logger?.WriteLine("Executing publish command");
+            if (dotnetCli.Publish(projectLocation, publishLocation, targetFramework, configuration, publishOptions) != 0)
+            {
+                throw new ElasticBeanstalkExceptions("Error executing \"dotnet publish\"", ElasticBeanstalkExceptions.CommonErrorCode.DotnetPublishFailed);
+            }
+
+            SetupAWSDeploymentManifest(publishLocation);
+
+            var zipArchivePath = Path.Combine(Directory.GetParent(publishLocation).FullName, new DirectoryInfo(projectLocation).Name + "-" + DateTime.Now.Ticks + ".zip");
+
+            this.Logger?.WriteLine("Zipping up publish folder");
+            Utilities.ZipDirectory(this.Logger, publishLocation, zipArchivePath);
+            this.Logger?.WriteLine("Zip archive created: " + zipArchivePath);
+
+            return Task.FromResult(true);
+        }
+
+        private void SetupAWSDeploymentManifest(string publishLocation)
```

Comment:
This is a duplicate of what is in DeployEnvironmentCommand. Create an EBUtilities class in the parent namespace and move this code to that class. That would make it consistent with the Lambda commands.

Reply:
done in the last commit - the function is not static, so I had to send a few parameters in to make it static